terraform/gcp/network.tf

```diff
@@ -0,0 +1,58 @@
+/**
+* Networking to support private clusters
+*
+* This config is only deployed when the enable_private_cluster variable is set
+* to true
+*/
+
+data "google_compute_network" "default_network" {
+  name    = "default"
+  project = var.project_id
+}
+
+data "google_compute_subnetwork" "default_subnetwork" {
+  name    = "default"
+  project = var.project_id
+  region  = var.region
+}
+
+resource "google_compute_firewall" "iap_ssh_ingress" {
+  count = var.enable_private_cluster ? 1 : 0
+
+  name    = "allow-ssh"
+  project = var.project_id
+  network = data.google_compute_network.default_network.name
+
+  allow {
+    protocol = "tcp"
+    ports    = ["22"]
+  }
+
+  // This range contains all IP addresses that IAP uses for TCP forwarding.
+  // https://cloud.google.com/iap/docs/using-tcp-forwarding
+  source_ranges = ["35.235.240.0/20"]
+}
+
+resource "google_compute_router" "router" {
```

Comment:
I tried to see if there exists a default router we can use too, and based on reading the docs and running `gcloud compute routers list` I've come to the conclusion we have to create our own. So this LGTM